### PR TITLE
Add optional credentials_file_path input for setup-gcloud

### PR DESCRIPTION
**Pull request by bricker**
Fixes GH-144

- Adds a new optional input to `setup-gcloud` which allows the caller to override the path at which the credentials are saved on disk. If not provided, the previous behavior (using `GITHUB_WORKSPACE`) is maintained.
- Adds tests for this action

### CODE REVIEW

**Comment by bricker** [comment]
@sethvargo How can I get this reviewed and merged?

**Comment by averikitsch** [comment]
Hi @bricker, I'll review this shortly. I want to note that we are working on a new process for merging external contributions.

File: setup-gcloud/tests/setup-gcloud.test.ts [hunk 1]
```diff
@@ -0,0 +1,211 @@
...
+  });
+
+  it('installs the gcloud SDK if it is not already installed', async function() {
+    this.stubs.isInstalled.returns(false);
+    this.stubs.getInput.withArgs('version').returns('888');
+    await run();
+    expect(this.stubs.installGcloudSDK.withArgs('888').callCount).to.eq(1);
+  });
+
+  it('uses the cached gcloud sdk if it was already installed', async function() {
```
Comment (averikitsch) [inline setup-gcloud/tests/setup-gcloud.test.ts:101]:
```suggestion
  it('uses the cached gcloud SDK if it was already installed', async function() {
```

**Review by averikitsch — APPROVED** [review]

File: setup-gcloud/dist/index.js [hunk 2]
```diff
@@ -34,7 +34,7 @@ module.exports =
 /******/ 	// the startup function
```
[the remark this answers is not in the document]
Comment (bricker) [reply setup-gcloud/dist/index.js:34]: This file is [built and added in a pre-commit hook](https://github.com/GoogleCloudPlatform/github-actions/blob/master/setup-gcloud/package.json#L13) (via Husky). Do you want me to delete this file, or just revert my changes? And should I remove the git hook too?

[the remark this answers is not in the document]
Comment (averikitsch) [reply setup-gcloud/dist/index.js:34]: The pre-commit hook has been removed. Please revert the changes to the `dist/index.js`. We will be generating this file as a part of our release process.

**Comment by averikitsch** [comment]
merging into a testing branch

**Pull request merged**